src/components/LoginButton.jsx

```diff
@@ -1,37 +1,59 @@
 import React, { useEffect, useState } from 'react';
 import { Link } from 'react-router-dom';
 import firebase from 'firebase';
+import { CircularProgress, makeStyles } from '@material-ui/core';
 import Button from '@material-ui/core/Button';
 import ProfileMenu from './ProfileDropdown';
 
+const useStyles = makeStyles(() => ({
+  progress: {
+    display: 'block',
+    margin: 'auto',
+  },
+}));
+
 export default function LoginButton() {
   const [isSignedIn, setIsSignedIn] = useState(!!firebase.auth().currentUser);
+  const [loading, setLoading] = useState(true);
+  const [error] = useState();
+  const classes = useStyles();
+
+  function updateUserStatus(user) {
+    setLoading(false);
+    setIsSignedIn(!!user);
+  }
 
   useEffect(
-    () => firebase.auth().onAuthStateChanged((user) => setIsSignedIn(!!user)),
+    () => firebase.auth().onAuthStateChanged((user) => updateUserStatus(user)),
     []
   );
 
   const signUserOut = () => firebase.auth().signOut();
   const user = firebase.auth().currentUser;
 
-  return isSignedIn ? (
-    <ProfileMenu
-      signOut={signUserOut}
-      name={user.displayName}
-      email={user.email}
-      photo={user.photoURL}
-    />
-  ) : (
-    <Button
-      variant="contained"
-      color="primary"
-      component={Link}
-      to={{
-        state: { showLoginDialog: true },
-      }}
-      replace>
-      Login
-    </Button>
+  return (
+    <>
+      {error?.toString() ||
+        (loading && <CircularProgress className={classes.progress} />) ||
+        (isSignedIn ? (
+          <ProfileMenu
+            signOut={signUserOut}
+            name={user.displayName}
+            email={user.email}
+            photo={user.photoURL}
+          />
+        ) : (
+          <Button
+            variant="contained"
+            color="primary"
+            component={Link}
+            to={{
+              state: { showLoginDialog: true },
+            }}
+            replace>
+            Login
+          </Button>
```

Comment:
When it re directs the first time user logs in there is a movement of the links due to size of avatar should this be concerning? As I fixed the issue with the button/avatar falling under the links

Reply:
I think that's what the [first](https://github.com/beyondhb1079/s4us/issues/42#issuecomment-766661364) issue is referring to.

Reply:
You can fix the shift issue by setting a big enough `minWidth` to the `<Grid item>` holding the button/profile menu in `HeaderNavMenu.jsx` and making the items in the item right-aligned.